lib/rucio/client/ruleclient.py

```diff
@@ -203,15 +203,19 @@ def approve_replication_rule(self, rule_id: str):
         exc_cls, exc_msg = self._get_exception(headers=r.headers, status_code=r.status_code, data=r.content)
         raise exc_cls(exc_msg)
 
-    def deny_replication_rule(self, rule_id: str):
+    def deny_replication_rule(self, rule_id: str, reason: Optional[str] = None):
         """
         :param rule_id:             Rule to be denied.
+        :param reason:              Reason for denying the rule.
         :raises:                    RuleNotFound
         """
 
         path = self.RULE_BASEURL + '/' + rule_id
         url = build_url(choice(self.list_hosts), path=path)
-        data = dumps({'options': {'approve': False}})
+        options: dict[str, Union[bool, str]] = {'approve': False}
```

Comment:
Rucio is still officially supporting Python 3.6. Consequently, you can’t use `dict` here and will have to use `typing.Dict` instead.

Reply:
Thanks! I can squash the commits to meaningful ones, once the CI is happy.

Reply:
Yes, please do. A single commit is expected for this PR.

Reply:
Alright, squashed (and rebased again), now that the CI succeeds.